lib/rubocop/cop/lint/nested_percent_literal.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for nested percent literals.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   # The percent literal for nested_attributes is parsed as four tokens,
+      #   # yielding the array [:name, :content, :"%i[incorrectly", :"nested]"].
+      #   attributes = {
+      #     valid_attributes: %i[name content],
+      #     nested_attributes: %i[name content %i[incorrectly nested]]
+      #   }
+      class NestedPercentLiteral < Cop
+        include PercentLiteral
+
+        MSG = 'Do not nest percent literals.'.freeze
```

Comment:
[IMO] I think this message is not good, this seems to mention to style problems. 
Users need more descriptive message.
For example:

https://github.com/bbatsov/rubocop/blob/6cc91fc8f0305cff3a3182934cf4b786c7f619dc/lib/rubocop/cop/lint/percent_string_array.rb#L30-L31

Reply:
Done (next commit).